src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final String TITLE_LINE = "Report for period";
+    private static final int DATA_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int WORKING_OUR_INDEX = 2;
+    private static final int INCOME_INDEX = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        return stringBuilder.append(TITLE_LINE)
+                .append(" " + dateFrom + " - " + dateTo)
+                .append(getData(names, data, dateFrom, dateTo)).toString();
+    }
+
+    private String getData(String[] names, String[] data, String dateFrom, String dateTo) {
```

Comment:
You can do all your logic in one 'getSalaryInfo' method. You will avoid creation of two StringBuilders